ocaml/idl/json_backend/gen_json.ml

```diff
@@ -381,6 +381,32 @@ end = struct
     else
       cmp
 
+  let rec list_dedup cmp =
+    let rec loop acc = function
+      | [] ->
+          List.rev acc
+      | [item] ->
+          loop (item :: acc) []
+      | a :: b :: rest when cmp a b = 0 ->
+          loop (a :: acc) rest
+      | a :: b :: rest ->
+          loop (a :: acc) (b :: rest)
+    in
+    loop []
+
+  let remove_outdated changes =
+    (* When several lifecycles transitions for the same entity, keep the most
+       significative one and drop the rest *)
+    changes
+    |> List.sort (fun ((_, nam_a, _, _) as a) ((_, nam_b, _, _) as b) ->
+           let cmp = String.compare nam_a nam_b in
```

Comment:
I am not sure I understand this logic: when the names are identical, call `compare_changes`, which again uses the complete structure. Is the purpose here to compare the two structures by element and controlling the order? I've used the code below in such a situation where I define an order by comparing elements in a particular order.

```
  let (<?>) a b =
    if a = 0 then b else a

  let compare a b =
        compare a.hour    b.hour
    <?> compare a.minutes b.minutes
    <?> compare a.seconds b.seconds
    <?> 0
```

Reply:
It orders the list of entity changes in a way that exploits the ordering of the type of changes (removed last, prototyped first).

Once that is done it drops items with a name that's the same as the preceding one. This way only the last change per entity is kept. For example if in a  release a field got deprecated and removed, only the removed change is kept.